mountpoint-s3/src/inode.rs

```diff
@@ -1211,6 +1284,84 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_forget() {
+        let superblock = Superblock::new("test_bucket", &Default::default(), Default::default());
+        let ino = 42;
+        let inode_name = "made-up-inode";
+        let inode = Inode {
+            inner: Arc::new(InodeInner {
+                ino,
+                parent: ROOT_INODE_NO,
+                name: inode_name.to_owned(),
+                full_key: inode_name.to_owned(),
+                kind: InodeKind::File,
+                sync: RwLock::new(InodeState {
+                    write_status: WriteStatus::Remote,
+                    stat: InodeStat::for_file(0, OffsetDateTime::now_utc(), None, Default::default()),
+                    kind_data: InodeKindData::File {},
+                    lookup_count: 5,
+                }),
+            }),
+        };
+        superblock.inner.inodes.write().unwrap().insert(ino, inode.clone());
+
+        superblock.forget(ino, 3);
+        let lookup_count = {
+            let inode_state = inode.inner.sync.read().unwrap();
+            inode_state.lookup_count
+        };
+        assert_eq!(lookup_count, 2, "lookup should have been reduced");
+        assert!(
+            superblock.inner.get(ino).is_ok(),
+            "inode should be present in superblock"
+        );
+
+        superblock.forget(ino, 2);
+        let lookup_count = {
+            let inode_state = inode.inner.sync.read().unwrap();
+            inode_state.lookup_count
+        };
+        assert_eq!(lookup_count, 0, "lookup should have been reduced");
+        assert!(
+            superblock.inner.inodes.read().unwrap().get(&ino).is_none(),
+            "inode should not be present in superblock"
+        );
+    }
+
+    #[tokio::test]
+    async fn test_forget_can_remove_inodes() {
```

Comment:
Maybe also a test that `readdir` (the non-plus version) inodes are never added to the superblock?